### PR TITLE
[4.0] Permissions custom element improvements

### PR DESCRIPTION
**Pull request by C-Lodder**
### Summary of Changes

Rework the Permissions Custom Element

- Cache variables in the constructor
- Decouple FontAwesome
- Make use of identical operators
- Eslint fixes

### Testing Instructions

1. To go the `permissions` tab in Global Configuration
2. Change the permission for different actions

### Expected result

- Refresh and ensure the permission you selected is still set
- Ensure the icons are visible

### CODE REVIEW

File: build/media/webcomponents/js/field-permissions/field-permissions.js [hunk 1]
```diff
@@ -1,152 +1,165 @@
...
+   * Lifecycle
+   */
+  connectedCallback() {
+    this.buttons = [].slice.call(document.querySelectorAll(`[${this.buttonDataSelector}]`));
+    if (this.buttons) {
+      this.buttons.forEach((button) => {
+        button.addEventListener('change', this.onDropdownChange);
+      });
+    }
+  }
```
Comment (dgrammatiko) [inline build/media/webcomponents/js/field-permissions/field-permissions.js]: Can you add also the disconnectedCallback like:
```js
          /**
	   * Lifecycle
	   */
	  disconnectedCallback() {
	    if (this.buttons) {
	      this.buttons.forEach((button) => {
	        button.removeEventListener('change', this.onDropdownChange);
	      });
	    }
	  }
```

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

File: build/media/webcomponents/js/field-permissions/field-permissions.js (hunk 1 above)
Comment (C-Lodder) [reply build/media/webcomponents/js/field-permissions/field-permissions.js]: oops, forgot about that. done

**Comment by brianteeman** [comment]
Shouldnt these be done upstream in the webcomponents repo?

**Comment by C-Lodder** [comment]
@brianteeman As far as I know there are 2 types of custom elements. Those that are replacements for Bootstrap components and live in a standalone repo, and those that are part of Joomla core that live in this repo.

**Comment by brianteeman** [comment]
Thanks for the clarification

**Comment by anuragteapot** [comment]
I have tested this item :white_check_mark: successfully on 20feb78d40b75711b9ac96dcce615e1c93e455fe<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23729">issues.joomla.org/tracker/joomla-cms/23729</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**